Core/include/Acts/EventData/detail/TestSourceLink.hpp

```diff
@@ -89,11 +90,19 @@ struct TestSourceLink final {
   constexpr std::size_t index() const { return sourceId; }
 
   struct SurfaceAccessor {
-    const Acts::TrackingGeometry& trackingGeometry;
+    const Acts::TrackingGeometry* trackingGeometry = nullptr;
+    const Acts::Experimental::Detector* detector = nullptr;
 
     const Acts::Surface* operator()(const Acts::SourceLink& sourceLink) const {
       const auto& testSourceLink = sourceLink.get<TestSourceLink>();
-      return trackingGeometry.findSurface(testSourceLink.m_geometryId);
```

Comment:
Shall we harmonise "Detector" and "TrackingGeometry" to both have 'findSurface(GeometryId)' - then this could be (temporarily) a template ?

Reply:
We have this pattern everywhere and it is actually the most common source why we - at the moment - need to mix Gen1 and Gen2 geometry: it's the lookup, probably we should solve that once and then re-use it.

(Also in the examples this will be needed for testing Gen2 navigation + simulation)

Reply:
`const Surface* Detector::findSurface(const GeometryContetxt& )` would be my suggestion.

Reply:
Closing this discussion here, as it is moved to the #3445